Core/Panel/README.md

```diff
@@ -0,0 +1,177 @@
+  environment (ENV_NAME) to avoid conflicts with other examples.
+
+
+## Publishing your app
+
+1. Ensure you are logged in (via ``docker login``) to DockerHub, Quay.io, or
+   another Docker registry.
+
+2. Run ``python -m ci publish`` to push your app's image to the registry.
+
+Once you have published your Docker image, you are ready to register that
+version of your app with Edge.
+
+As a quick reminder, in Edge, there is a distinction between an _app_ and an
+_app version_.  Your organization administrator can define an _app_, and then
+developers are free to register _app versions_ associated with that app.  This
+ensures that org administrators have full control over what shows up on the
+dashboard, but gives developers freedom to publish new versions by themselves.
+
+The easiest way to register a new app version is by logging in to Edge, going
+to the gear icon in the upper-right corner, and navigating to the app in
+question. There is a form to fill out which asks for the version, Quay.io or
+DockerHub URL for the image, and some other information.
+
+It is also possible to register app versions programmatically, for example
+from within a GitHub Actions workflow.  See the example at the end of this
+README.
+
+
+## Getting EdgeSession to work in development
+
+When you run your app on Edge, you can create an EdgeSession object simply by
+calling the constructor (``mysession = EdgeSession()``).  This works by
+collecting environment variables set by Edge when the container is launched.
+
+When developing locally, it's also convenient to have an EdgeSession.  You
+can get the "ci" module to inject the appropriate environment variables, so
+that your ``EdgeSession()`` call will work with ``python -m ci run`` and
+``python -m ci preflight``.  
+
+To do so, follow this procedure:
+
+* Copy the "dev_settings.json.example" file to "dev_settings.json".
+* Define EDGE_API_SERVICE_URL in that file.  The typical value here is
+  ``"https://edge.enthought.com/services/api"``.
+* Define EDGE_API_ORG in that file.  This is the "short name" displayed in
+  the URL bar when you log into an organization, for example, ``"default"``.
+* Define EDGE_API_TOKEN.  You can get one of these by going to the
+  ``/hub/token`` endpoint on the Edge server.
+
+Be sure *not* to check the "dev_settings.json" file into source control, as it
+contains your API token.
+
+
+## Viewing console output
+
+When running with ``python -m ci run``, the app's output will be displayed
+on the console where you launched it.
+
+## Guidelines for your Dockerfile
+
+Edge will run your app next to a built-in reverse proxy, which allows
+you to skip a lot of work in the development process.  This includes stripping
+the prefix from requests, handling the OAuth2 login flow, pinging JupyterHub
+for container activity, and more.  But, there are a few guidelines you will
+need to follow in your own Dockerfile.
+
+* Don't change the user to anything other than ``app`` (for example, by the 
```

Comment:
I think this doesn't apply anymore, since it only existed to keep edge-native-base happy.